TKGame/Weapons/Sword.cs

```diff
@@ -1,63 +1,67 @@
+using TKGame.Components.Interface;
 
 namespace TKGame.Content.Weapons
 {
     class Sword : Weapon
     {
-        static Sword instance;
-        private static object syncRoot = new object();
-        /// <summary>
-        /// Locks single sword instance
-        /// </summary>
-        public static Sword Instance
-        {
-            get
-            {
-                if (instance == null)
-                {
-                    lock (syncRoot)
-                    {
-                        if (instance == null)
-                        {
-                            instance = new Sword();
-                        }
-                    }
-                }
-                return instance;
-            }
-        }
+        private static readonly int WALK_ACCELERATION = 1;
+        private static readonly int JUMP_HEIGHT = -1;
+        private float MOVEMENT_SPEED = 500f;
+
+        ////Components
+        //InputComponent input;
+        //PhysicsComponent physics;
+        //GraphicsComponent graphics;
+
 
         /// <summary>
         /// Private Constructor for Sword
         /// </summary>
-        private Sword()
+        public Sword()
         {
+            //input = input_;
+            //physics = physics_;
+            //graphics = graphics_;
             weaponTexture = Art.WeaponTexture;
-            position = new Vector2(0, 0);
-            
+            position = new Vector2(815, 730);
+            weaponRect = new Rectangle(815, 730, 85, 85);
+            damageStat = 1;
         }
 
-        public void Update(GameTime gameTime, Vector2 position, Vector2 velocity)
+        public override void Draw(SpriteBatch spriteBatch)
         {
 
+            weaponTexture = Art.WeaponTexture;
+            //spriteBatch.Draw(weaponTexture, position, weaponRect, color, 0, position, 0.2f, orientation, 0);
+            //spriteBatch.Draw(weaponTexture, weaponRect, color);
         }
 
-        public void Draw(SpriteBatch spriteBatch)
+        public override int DamageEnemy(Weapon weapon)
         {
-            spriteBatch.Draw(weaponTexture, position, null, color, 0, size / 4f, 1f, orientation, 0);
+            throw new NotImplementedException();
         }
 
-
-        public override int DamageEnemy(Weapon weapon)
+        public override void Update(Weapon w)
         {
-            throw new NotImplementedException();
+            if (Input.KeyboardState.IsKeyDown(Keys.A))
+                w.velocity.X = -WALK_ACCELERATION * MOVEMENT_SPEED;
+            else if (Input.KeyboardState.IsKeyDown(Keys.D))
+                w.velocity.X = WALK_ACCELERATION * MOVEMENT_SPEED;
+            else if (Input.KeyboardState.IsKeyDown(Keys.W))
+                w.velocity.Y = -WALK_ACCELERATION * MOVEMENT_SPEED;
+            else if (Input.KeyboardState.IsKeyDown(Keys.S))
+                w.velocity.Y = WALK_ACCELERATION * MOVEMENT_SPEED;
+            else
+                w.velocity = Vector2.Zero;
         }
     }
 }
```

Comment:
Why is the sword walking? If this is meant to keep the sword with the player's sprite this is a very odd solution. You should be able to query the Player's Instance and play the sprite relative to the player